third_party/mbedtls/mbedtls-config.h

```diff
@@ -699,6 +708,32 @@ __inline int windows_kernel_snprintf(char * s, size_t n, const char * format, ..
  */
 //#define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+/**
+ * Enables ciphersuite ECDHE ECDSA with x.509 certificate
+ * for Application Coap Secure.
+ */
+#ifndef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+//#define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+#if defined(MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED)
+
+/**
+ * to increase the security with ECDHE ECDSA enable
+ * ECDSA DETERMINISTIC and X509 CHECK KEY USAGE.
+ * Not required!
+ */
+#ifndef MBEDTLS_ECDSA_DETERMINISTIC
+//#define MBEDTLS_ECDSA_DETERMINISTIC
```

Comment:
Should this be enabled when `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is `1`?

Reply:
Yes, it should be (but must not) enabled, when `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is defined.

Reply:
Enabling `MBEDTLS_ECDSA_DETERMINISTIC` results in the following linker error:
```
Undefined symbols for architecture x86_64:
  "_mbedtls_hmac_drbg_free", referenced from:
      _mbedtls_ecdsa_sign_det in libmbedcrypto.a(libmbedcrypto_a-ecdsa.o)
  "_mbedtls_hmac_drbg_init", referenced from:
      _mbedtls_ecdsa_sign_det in libmbedcrypto.a(libmbedcrypto_a-ecdsa.o)
  "_mbedtls_hmac_drbg_random", referenced from:
      _mbedtls_ecdsa_sign_det in libmbedcrypto.a(libmbedcrypto_a-ecdsa.o)
  "_mbedtls_hmac_drbg_seed_buf", referenced from:
      _mbedtls_ecdsa_sign_det in libmbedcrypto.a(libmbedcrypto_a-ecdsa.o)
```

Reply:
Recognized.

Reply:
Fixed. I have removed `repo/library/hmac_drbg.c` from mbed's Makefile.am inadvertently.
Done.